packages/blockchain-wallet-v4-frontend/src/modals/Onboarding/CompleteProfile/index.tsx

```diff
@@ -107,6 +108,11 @@ class CompleteProfile extends PureComponent<Props, State> {
     this.props.buySellActions.fetchAccumulatedTrades({ product: ProductTypes.SIMPLEBUY })
 
     const { currentStep } = this.props.data
+    peeksheetViewed({
+      current_step_completed: String(currentStep),
+      device: '',
+      platform: ''
```

Comment:
This is unfortunate. Since we require `device` and `platform` on our segment events typescript enforces these. A possible solution for this is to make these optional in segment, knowing that we will always add them to the events anyways.

Reply:
we have those as fixed strings
```
device: 'WEB',
platform: 'WALLET'
```
might do the very same.

Reply:
possible to have some sort of wrapper that auto-adds these or maybe modify the typewriting script to make these optional to the types?

Reply:
@mitchmalinin and I were discussing how to do this today. My initial thought is that we somehow `import *` all the typewriter methods, do something like `type NewPeekSheetViewed = Omit<PeekSheetViewed, 'device' | 'platform'>` and then re-export them (or a wrapper function that calls them) with this new type so we can keep the all the other type goodness from typewriter. device and platform will get added to the actual event through Milan's `trackEvent` work before being sent up to our api.